### PR TITLE
Basic CLI for metrics-importer

### PR DESCRIPTION
**Pull request by camlorn**
This contains the following:

- Support for metrics-importer commands:
  - `metrics-importer logs` to get logs
  - `metrics-importer source check-query` to test a query
  - `metrics-importer importer create` to create importers.
- The beginnings of a properly abstracted HTTP client for the service.
- Support for catching errors from Morgue commands that use async/await directly that doesn't trigger Node's deprecation warning.

This is intended to be a minimal cut for CI and troubleshooting purposes with further improvements based on user demand.  Notable missing functionality includes the ability to list, modify, or delete resources.

I wrote this with an eye to being abstractable, reusable, and a pattern we may wish to follow with future microservices that need Morgue interfaces.  Though that abstraction hasn't happened, the intent is that it would be relatively easy to pull out routing and validation helpers and to follow this directory structure in future.  Notably, it is possible to use `lib/metricsImporter/client.js` as a library to control the service programmatically.  Though I wouldn't call that API stable at this time, the intent is that it won't change.

### CODE REVIEW

File: bin/morgue.js [hunk 1]
```diff
@@ -7461,7 +7471,22 @@ function main() {
       errx("Unable to read configuration: " + err.message + ".");
     }
 
-    command(argv, config);
+    /*
```
Comment (pkhuong) [inline bin/morgue.js:7474]: I'd split out that change in a different commit; it'll be easier to describe that way too.

File: lib/coroner.js [hunk 2]
```diff
@@ -594,6 +595,36 @@ CoronerClient.prototype.delete_objects = function(universe, project, objects, pa
...
+  if (!serviceEntry) {
+    throw new Error(`No ${ name } service is configured`);
+  }
+  const endpoint = serviceEntry.endpoint;
+  if (!endpoint) {
+    throw new Error(`Service ${ name } doesn't have an endpoint`);
+  }
+  /*
+   * Frontend gets to assume that relative urls will work because it's
+   * in a browser and pointed at the same domain. We can't, because this is
```
Comment (pkhuong) [inline lib/coroner.js:617]: do we ever expect to see `//` absolute URLs, without the scheme?

File: lib/metricsImporter/cli.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
+const client = require('./client');
+
+function validateOne(option, value) {
+  if (!value) {
+    console.error(`--${ option } is required`);
+    return false;
+  }
+  if (Array.isArray(value)) {
+    console.Error(`Specify only one -${ option }`);
```
Comment (pkhuong) [inline lib/metricsImporter/cli.js:9]: what's up with `console.error` VS `console.Error` ? I'd also be consistent in the general syntax of the error message. Maybe "--${ option } must have exactly one value"  ?

File: lib/metricsImporter/cli.js [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    return false;
+  }
+  return true;
+}
+
+function validateZeroOrOne(option, value) {
+  if (Array.isArray(value)) {
+    console.error(`SSpecify at most one --${ option }`);
```
Comment (pkhuong) [inline lib/metricsImporter/cli.js:17]: Typo: "SSpecify" should have only one S

File: lib/metricsImporter/cli.js [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    const project = args._[0];
+    const sourceId = args['source'];
+    const importerId = args['importer'];
+    let limit = 100;
+
+    if (!project) {
+      console.log("Project is required");
+      return;
+    }
+    if (Array.isArray(sourceId)) {
```
Comment (pkhuong) [inline lib/metricsImporter/cli.js:83]: Can we use the new `validateOne` / `validateZeroOrOne` helpers for consistent error messages?

File: lib/metricsImporter/cli.js [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+
+    if ('limit' in args) {
+      limit = Number.parseInt(args.limit);
+      if (Number.isNaN(limit) || limit <= 0) {
+        console.error("--limit must be a positive integer");
+        return;
+      }
+    }
+
+
```
Comment (pkhuong) [inline lib/metricsImporter/cli.js:104]: Do we have a styling tool for morgue? Might be worth running whatever roughly matches morgue style on this new file.

File: README.md [hunk 7]
```diff
@@ -1160,3 +1160,67 @@ Possible roles:
 If a user has access through multiple sources (e.g. they belong to
 two teams and also have direct project membership) they will have
 the highest privileges afforded by any of those access routes.
+
```
Comment (pkhuong) [inline README.md:1163]: We seem to only use one empty line before subsections.

File: README.md [hunk 8]
```diff
@@ -1160,3 +1160,67 @@ Possible roles:
...
+Determines if a query is valid by running it against a source as if it had been
+used with an importer and displays diagnostic information.  For example:
+
+```
+morgue metrics-importer source check-query --source my-source-uuid \
+--query 'select time, value from test where time >= $1 and time < $2'
+```
+
+
```
Comment (pkhuong) [inline README.md:1185]: and only one empty line here.

File: README.md [hunk 9]
```diff
@@ -1160,3 +1160,67 @@ Possible roles:
...
+### `importer create`
+
+Creates an importer.  Takes the following options:
+
+Option       |
```
Comment (pkhuong) [inline README.md:1190]: Does this line up correctly outside the diff view? I see one missing space after `Option`

File: README.md [hunk 10]
```diff
@@ -1160,3 +1160,67 @@ Possible roles:
...
+-------------- | --------------------------------------------------------------
+--source-    | UUUID of the source to associate the importer with.
```
Comment (pkhuong) [inline README.md:1192]: Typos: there's a stray `-` after `source`, and one too many `U` in `UUUID`. We also need one more space after `--source` to line things up.

File: lib/coroner.js (hunk 2 above)
Comment (camlorn) [reply lib/coroner.js:617]: I don't think so for the time being, and think it's perhaps best to wait for that to be a roblem before trying to resolve it.

File: lib/metricsImporter/cli.js (hunk 3 above)
Comment (camlorn) [reply lib/metricsImporter/cli.js:9]: It's a typo.

**Comment by camlorn** [comment]
@pkhuong 
I think I've addressed everything but formatting. We don't have a good formatting config for this as of yet.  Also, I can't link documentation because this doesn't exist internally at this time.

**Comment by pkhuong** [comment]
Looks good. I only marked a couple formatting fixes.

**Comment by camlorn** [comment]
Formatting changes are fixed.  Also, note that I forgot to commit package.json and package-lock.json; those are now part of this PR, should they need review.
